x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/LookupJoinTypesIT.java

```diff
@@ -0,0 +1,463 @@
+ *             "field_double": 1.0,  // this is mapped as type 'float' (a float with the name of the main index field)
+ *             "other": "value"
+ *         }
+ *     </pre></dd>
+ *     <dt>index_float_double</dt>
+ *     <dd>Index containing a single document with a field of type 'double' like: <pre>
+ *         {
+ *             "field_float": 1.0,  // this is mapped as type 'double' (a double with the name of the main index field)
+ *             "other": "value"
+ *         }
+ *     </pre></dd>
+ *     <dt>index_float_float</dt>
+ *     <dd>Index containing single document with a field of type 'float' like: <pre>
+ *         {
+ *             "field_float": 1.0,  // this is mapped as type 'float'
+ *             "other": "value"
+ *         }
+ *     </pre></dd>
+ *     <dt>index</dt>
+ *     <dd>Index containing document like: <pre>
+ *         {
+ *             "field_double": 1.0,  // this is mapped as type 'double'
+ *             "field_float": 1.0    // this is mapped as type 'float'
+ *         }
+ *     </pre></dd>
+ * </dl>
+ * Note that the lookup indexes have fields with a name that matches the type in the main index, and not the type actually used in the
+ * lookup index. Instead, the mapped type should be the type of the right-hand side of the pair being tested.
+ * Then we can run queries like:
+ * <pre>
+ *     FROM index | LOOKUP JOIN index_double_float ON field_double | KEEP other
+ * </pre>
+ * And assert that the result exists and is equal to "value".
+ */
+@ClusterScope(scope = SUITE, numClientNodes = 1, numDataNodes = 1)
+public class LookupJoinTypesIT extends ESIntegTestCase {
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(EsqlPlugin.class, MapperExtrasPlugin.class);
+    }
+
+    private static final Map<String, TestConfigs> testConfigurations = new HashMap<>();
+    static {
+        // Initialize the test configurations for string tests
+        {
+            TestConfigs configs = testConfigurations.computeIfAbsent("strings", TestConfigs::new);
+            configs.addPasses(KEYWORD, KEYWORD);
+            configs.addPasses(TEXT, KEYWORD);
+            configs.addFailsText(KEYWORD);
+            configs.addFailsText(TEXT);
+        }
+
+        // Test integer types
+        var integerTypes = List.of(BYTE, SHORT, INTEGER);
+        {
+            TestConfigs configs = testConfigurations.computeIfAbsent("integers", TestConfigs::new);
+            for (DataType mainType : integerTypes) {
+                for (DataType lookupType : integerTypes) {
+                    configs.addPasses(mainType, lookupType);
+                }
+                // Long is currently treated differently in the validation, but we could consider changing that
+                configs.addFails(mainType, LONG);
+                configs.addFails(LONG, mainType);
+            }
+        }
+
+        // Test float and double
+        var floatTypes = List.of(HALF_FLOAT, FLOAT, DOUBLE);
```

Comment:
We also support `scaled_float` and could already include it here.

Reply:
Great find!

I meant: `scaled_float` is supported by ESQL, and our docs don't say that you can't use `scaled_float` in `LOOKUP JOIN`, so we need to either test it to ensure that's correct, or consider such queries invalid and document the limitation.

I'd say we can go with the latter for now, and it's great to have confirmation for this type!

Reply:
OK. I investigated, and LOOKUP JOIN works fine on scaled floats. I did a bunch of manual tests in a running instance of elasticsearch, and the matches work as expected. I'm leaving it as supported and made sure we test a valid value here, with a non-unity scaling_factor.